crates/nu-command/tests/commands/into_int.rs

```diff
@@ -101,3 +101,54 @@ fn into_int_datetime(#[case] time_in: &str, #[case] int_out: &str) {
 
     assert_eq!(int_out, actual.out);
 }
+
+#[test]
+fn into_int_duration_with_unit() {
+    let actual = nu!("1hr | into int --unit sec");
```

Comment:
I'm a little bit confused. As with the current `into duration` implementation, the `--unit` parameter is supposed to indicate what unit the _input_ is in, not what unit to convert it _to_. For the convert to functionality there are the `format blah` commands.

So, this test would be `3600 | into int -unit sec` and the output would be `1hr`. The `sec` tells the command what unit the `3600` is in. And the filesize test below would be `1 | into int --unit MB`, the `MB` tells the command what `1`'s unit is.

Please help me understand if I'm missing something.

Reply:
As discussed in the issue, I was tring on implement what I think we agreed to https://github.com/nushell/nushell/issues/14469#issuecomment-2816828012.

That said, I completely agree that having a --unit flag with different behaviors (representing the units inputs are in, or units I want the output to be in), could lead to confusion

We can also just drop this proposal of mine and just keep Bahex's solution to simply use division... But that's why I asked for confirmation again in the issue before implementing the code...

Reply:
Sorry for being confusing and causing you extra work here. I think we should keep 1 usage of --unit and that is for input. Maybe it should be called --input-unit to be more clear and leave the formatting for the format commands or math ala bahex.

Reply:
I guess all this branch is not to throw away, but yeah I wish I could have avoided this extra work...

So if you want I can:
- remove the --unit option for the "into int" command
- No need to rename --unit to --input-unit in other commands
- Add an example in filesize and duration types showing how to easily convert between units using division

And we can keep the changes in unit.rs. As I wrote in the description, I wanted to improve the parsing of units, the conversions between them. I started this in this branch but this is not complete.

Reply:
Or I can just move out all the unit stuff into another branch and just merge this when it's all done

Reply:
i'm fine with keeping the refactor of the unit stuff and the examples like you suggest.

Reply:
Okay :)

I just pushed the removal of the --unit flag and I fixed the conflicts.

I don't know if adding examples in filesize and duration types showing how to to easily convert between units using division like Bahex said in necessary, though. I didn't know if here in the code was right, so I thought about adding them [here](https://www.nushell.sh/lang-guide/chapters/types/basic_types/duration.html), but I'm not sure.. Plus, the "format duration" command seems even more useful than this division trick. What do you think? Or do you see another place in the docs where this trick might fit?

I also add new tests for "format duration" and "format filesize"

Reply:
Updated the PR description just now